hack/verify-codecgen.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+
+cd "${KUBE_ROOT}"
+
+generated_files=`
+  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './_output' \
+        -o -wholename './release' \
+        -o -wholename './target' \
+        -o -wholename '*/third_party/*' \
+        -o -wholename '*/Godeps/*' \
+      \) -prune \
+    \) -name '*.generated.go'`
+
+# create a nice clean place to put codecgen there
+_tmpdir="$(mktemp -d -t codecgen.XXXXXX)"
+function cleanup {
+  echo "Removing ${_tmpdir}"
+  rm -rf "${_tmpdir}"
+}
+trap cleanup EXIT
+
+# build codecgen tool
+_gopath=${GOPATH}
+export GOPATH="${_tmpdir}"
+go get -u github.com/ugorji/go/codec/codecgen 2>/dev/null
+go install github.com/ugorji/go/codec/codecgen 2>/dev/null
+CODECGEN="${_tmpdir}/bin/codecgen"
+export GOPATH=${_gopath}
```

Comment:
I assume all this hackery is because of Travis? Could you leave some comments explaining why this is here?